### ISSUE TITLE
最近拜读了您的论文“Generalization Matters: Loss Minima Flattening via Parameter Hybridization for Efficient Online Knowledge Distillation”之后，对我有非常大的启发。但是在阅读了开源代码之后，我有些疑问，您的代码中fusion方法得到的net好像并没有使用啊，您能说明下吗？谢谢了。

### ISSUE THREAD

**Issue by sewen1998** [tianlizhang/OKDPH#1]

**Comment by tianlizhang** [comment]
If you modify the state of the dictionary of the PyTorch model, the model stored in the list will also be modified. When you add a PyTorch model to a list, you’re essentially adding a reference to that model object. So any changes made to the original object will also affect all references to it.
Here’s an example to illustrate this:
```
import torch.nn as nn
model = nn.Sequential(
    nn.Linear(10, 5),
    nn.ReLU(),
    nn.Linear(5, 1)
)
model_list = []
model_list.append(model)

model.state_dict()['0.weight'][0][0] = 100

print(model_list[0].state_dict()['0.weight'][0][0])
```

As you can see, modifying the state dict of the model also affects the model in the list.

**Comment by sewen1998** [comment]
Thank you very much!!!




sewen
***@***.***



&nbsp;




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "tianlizhang/OKDPH"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2023年6月10日(星期六) 晚上11:12
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [tianlizhang/OKDPH] 最近拜读了您的论文“Generalization Matters: Loss Minima Flattening via Parameter Hybridization for Efficient Online Knowledge Distillation”之后，对我有非常大的启发。但是在阅读了开源代码之后，我有些疑问，您的代码中fusion方法得到的net好像并没有使用啊，您能说明下吗？谢谢了。 (Issue #1)





 
If you modify the state of the dictionary of the PyTorch model, the model stored in the list will also be modified. When you add a PyTorch model to a list, you’re essentially adding a reference to that model object. So any changes made to the original object will also affect all references to it.
 Here’s an example to illustrate this:
 import torch.nn as nn model = nn.Sequential(     nn.Linear(10, 5),     nn.ReLU(),     nn.Linear(5, 1) ) model_list = [] model_list.append(model) model.state_dict()['0.weight'][0][0] = 100 print(model_list[0].state_dict()['0.weight'][0][0])  
As you can see, modifying the state dict of the model also affects the model in the list.

**Comment by lyhwyl** [comment]
最近阅读了您的这篇‘Generalization Matters: Loss Minima Flattening via Parameter Hybridization for Efficient Online Knowledge Distillation’论文，对于这篇论文，我深受启发，于是进行了复现。在复现过程中，因为imagenet数据集比较庞大，于是我使用了Tiny-imagenet数据集（我已经将其格式和imagenet数据集的格式进行了调整）进行了复现，所以我想请问您在使用这个数据集上进行复现实验时，是否设置了一些参数还是修改了什么地方，如果能告知，我将不胜感谢。

**Comment by sewen1998** [comment]
这篇论文的作者不是我啊，我也是求助人




sewen1998
***@***.***



&nbsp;




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "tianlizhang/OKDPH"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2024年3月11日(星期一) 中午11:21
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [tianlizhang/OKDPH] 最近拜读了您的论文“Generalization Matters: Loss Minima Flattening via Parameter Hybridization for Efficient Online Knowledge Distillation”之后，对我有非常大的启发。但是在阅读了开源代码之后，我有些疑问，您的代码中fusion方法得到的net好像并没有使用啊，您能说明下吗？谢谢了。 (Issue #1)





 
最近阅读了您的这篇‘Generalization Matters: Loss Minima Flattening via Parameter Hybridization for Efficient Online Knowledge Distillation’论文，对于这篇论文，我深受启发，于是进行了复现。在复现过程中，因为imagenet数据集比较庞大，于是我使用了Tiny-imagenet数据集（我已经将其格式和imagenet数据集的格式进行了调整）进行了复现，所以我想请问您在使用这个数据集上进行复现实验时，是否设置了一些参数还是修改了什么地方，如果能告知，我将不胜感谢。
